reference-architecture/azure-ansible/allinone.json

```diff
@@ -0,0 +1,670 @@
+              "subnet" : {
+                "id" : "[variables('subnetRef')]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion" : "2017-04-01",
+      "type" : "Microsoft.Network/publicIPAddresses",
+      "name" : "[variables('publicIPAddressName')]",
+      "location" : "[resourceGroup().location]",
+      "properties" : {
+        "publicIPAllocationMethod" : "Static",
+        "dnsSettings" : {
+          "domainNameLabel" : "[resourceGroup().name]"
+        }
+      }
+    },
+    {
+      "apiVersion" : "[variables('apiVersion')]",
+      "type" : "Microsoft.Network/virtualNetworks",
+      "name" : "[variables('virtualNetworkName')]",
+      "location" : "[variables('location')]",
+      "tags" : {
+        "displayName" : "VirtualNetwork"
+      },
+      "properties" : {
+        "addressSpace" : {
+          "addressPrefixes" : [
+            "[variables('addressPrefix')]"
+          ]
+        },
+        "subnets" : [
+          {
+            "name" : "[variables('virtualNetworkName')]",
+            "properties" : {
+              "addressPrefix" : "[variables('addressPrefix')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "type" : "Microsoft.Compute/virtualMachines/extensions",
+      "name" : "[concat(resourceGroup().name,'/installcustomscript')]",
+      "apiVersion" : "[variables('apiVersion')]",
+      "location" : "[resourceGroup().location]",
+      "tags" : {
+        "displayName" : "VirtualMachineCustomScriptExtension"
+      },
+      "dependsOn" : [
+        "[concat('Microsoft.Compute/virtualMachines/', resourceGroup().name)]"
+      ],
+      "properties" : {
+        "publisher" : "Microsoft.Azure.Extensions",
+        "type" : "CustomScript",
+        "typeHandlerVersion" : "2.0",
+        "autoUpgradeMinorVersion" : true,
+        "settings" : {
+          "fileUris" : [
+            "[concat(variables('baseTemplateUrl'), 'allinone.sh')]"
+          ]
+        },
+        "protectedSettings" : {
+          "commandToExecute" : "[ concat('bash allinone.sh ', resourceGroup().name, ' ', resourceGroup().name, ' ',  parameters('adminUsername'), ' ', parameters('adminPassword'), ' ', reference(variables('publicIPAddressName')).dnsSettings.fqdn, ' ', '0', ' ', reference('onevmPublicIP').ipAddress, ' ', parameters('RHNUserName'), ' ', parameters('RHNPassword'),' ', parameters('SubscriptionPoolId'), ' ',  variables('sQuote'),  parameters('sshPrivateData'), variables('sQuote'),  ' ', variables('sQuote'), parameters('sshKeyData'), variables('sQuote'),' ',variables('registryStorageName'),' ',listKeys(resourceId('Microsoft.Storage/storageAccounts',variables('registryStorageName')),'2015-06-15').key1,' ',variables('location'),' ',variables('subscriptionid'),' ',variables('tenantid'),' ',parameters('aadclientid'),' ',parameters('aadclientsecret'),' ', variables('rhsmmode'),' ',parameters('metrics'),' ',parameters('logging'),' ',parameters('opslogging'))]"
```

Comment:
its intentional to keep the args aligned and consistent. Yes on quote as it needed to protect from getting command line fun on the allinone.sh